src/Storages/MeiliSearch/StorageMeiliSearch.cpp

```diff
@@ -0,0 +1,191 @@
+        if (node->children.size() == 1) {
+            return node->children[0];
+        } else {
+            return nullptr;
+        }
+    }
+    for (const auto& next : node->children) {
+        auto res = getFunctionParams(next, name);
+        if (res != nullptr) {
+            return res;
+        }
+    }
+    return nullptr;
+}
+
+Pipe StorageMeiliSearch::read(
+    const Names & column_names,
+    const StorageMetadataPtr & metadata_snapshot,
+    SelectQueryInfo & query_info,
+    ContextPtr /*context*/,
+    QueryProcessingStage::Enum /*processed_stage*/,
+    size_t max_block_size,
+    unsigned)
+{
+    metadata_snapshot->check(column_names, getVirtuals(), getStorageID());
+
+    ASTPtr original_where = query_info.query->clone()->as<ASTSelectQuery &>().where();
+    ASTPtr query_params = getFunctionParams(original_where, "meiliMatch");
+    
+
+    std::unordered_map<String, String> kv_pairs_params;
+    if (query_params) {
+        LOG_TRACE(log, "Query params: {}", convertASTtoStr(query_params));
+        for (const auto& el : query_params->children) {
+            auto str = el->getColumnName();
+            auto it = find(str.begin(), str.end(), '=');
+            if (it == str.end()) {
+                throw Exception(
+                    "meiliMatch function must have parameters of the form \'key=value\'",
+                    ErrorCodes::BAD_QUERY_PARAMETER);
+            }
+            String key(str.begin() + 1, it);
+            String value(it + 1, str.end() - 1);
+            kv_pairs_params[key] = value;
+        }
+    } else {
+        LOG_TRACE(log, "Query params: none");
+    }
+
+    for (const auto& el : kv_pairs_params) {
+        LOG_TRACE(log, "Parsed parameter: key = " + el.first + ", value = " + el.second);
+    }
+
+    Block sample_block;
+    for (const String & column_name : column_names)
+    {
+        auto column_data = metadata_snapshot->getColumns().getPhysical(column_name);
+        sample_block.insert({ column_data.type, column_data.name });
+    }
+
+    return Pipe(std::make_shared<MeiliSearchSource>(config, sample_block, max_block_size, 0, kv_pairs_params));
+}
+
+SinkToStoragePtr StorageMeiliSearch::write(
+    const ASTPtr & /*query*/, 
+    const StorageMetadataPtr& metadata_snapshot, 
+    ContextPtr /*local_context*/)
+{
+    LOG_TRACE(log, "Trying update index: " + config.index);
+    return std::make_shared<SinkMeiliSearch>(
+        config,
+        metadata_snapshot->getSampleBlock(),
+        20000);
+}
+
+MeiliSearchConfiguration getConfiguration(ASTs engine_args) 
+{
+
+    if (engine_args.size() != 3) {
+        throw Exception(
+                "Storage MeiliSearch requires 3 parameters: MeiliSearch('url/host:port', index, key).",
+                ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);
+    }
+
+    // 7700 - default port
+    try {
+        auto parsed_host_port = parseAddress(engine_args[0]->as<ASTLiteral &>().value.safeGet<String>(), 7700);
+
```

Comment:
try catch is not a good way to check argument pattern.
Better to check arguments size and if 4 then it will mean host port are passed separately and if 3 then it is just url.
But if there is a possibility that more arguments will be added, then it will be very inconvenient, though this is exactly how it is done in `TableFunctionS3.cpp`, so probably be will be Ok.